backend/src/main/java/ai/verta/modeldb/entities/versioning/RepositoryEntity.java

```diff
@@ -1,54 +1,58 @@
 package ai.verta.modeldb.entities.versioning;
 
+import ai.verta.modeldb.ModelDBConstants;
 import ai.verta.modeldb.dto.WorkspaceDTO;
+import ai.verta.modeldb.entities.AttributeEntity;
+import ai.verta.modeldb.utils.RdbmsUtils;
 import ai.verta.modeldb.versioning.Repository;
 import ai.verta.modeldb.versioning.Repository.Builder;
 import ai.verta.modeldb.versioning.RepositoryAccessModifierEnum.RepositoryAccessModifier;
-import ai.verta.modeldb.versioning.RepositoryVisibilityEnum.RepositoryVisibility;
 import ai.verta.modeldb.versioning.SetRepository;
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Objects;
 import java.util.Set;
+import javax.persistence.CascadeType;
 import javax.persistence.Column;
 import javax.persistence.Entity;
 import javax.persistence.GeneratedValue;
 import javax.persistence.GenerationType;
 import javax.persistence.Id;
 import javax.persistence.ManyToMany;
+import javax.persistence.OneToMany;
 import javax.persistence.OrderBy;
 import javax.persistence.Table;
+import org.hibernate.annotations.LazyCollection;
+import org.hibernate.annotations.LazyCollectionOption;
 
 @Entity
 @Table(name = "repository")
 public class RepositoryEntity {
 
   public RepositoryEntity() {}
 
-  public RepositoryEntity(
-      String name,
-      WorkspaceDTO workspaceDTO,
-      String owner,
-      RepositoryVisibility repositoryVisibility,
-      RepositoryAccessModifier repositoryAccessModifier) {
-    this.name = name;
+  public RepositoryEntity(Repository repository, WorkspaceDTO workspaceDTO)
+      throws InvalidProtocolBufferException {
+    this.name = repository.getName();
     this.date_created = new Date().getTime();
     this.date_updated = new Date().getTime();
-    if (repositoryVisibility != null) {
-      this.repository_visibility = repositoryVisibility.getNumber();
-    }
-    if (repositoryAccessModifier != null) {
-      this.repositoryAccessModifier = repositoryAccessModifier.getNumber();
-    }
+    this.repository_visibility = repository.getRepositoryVisibilityValue();
+    this.repositoryAccessModifier = repository.getRepositoryAccessModifierValue();
```

Comment:
not sure about these changes

Reply:
Instead of sending each repository field as a parameter, I have sent a repository object as a parameter so it will never be null. it will have default value.